OpenSim/Simulation/Test/testForces.cpp

```diff
@@ -782,6 +774,131 @@ void testFunctionBasedBushingForce()
+    
+    osimModel->addBody(&ball);
+    osimModel->addJoint(&sliderY);
+    
+    // Create base body and attach it to ground with a weld
+
+    OpenSim::Body base("base_body", mass, Vec3(0), mass*SimTK::Inertia::sphere(0.1));
+    base.addMeshGeometry("sphere.vtp");
+    base.scale(Vec3(ball_radius), false);
+    
+    WeldJoint weld("", ground, Vec3(0), Vec3(0), base, Vec3(0), Vec3(0));
+    osimModel->addBody(&base);
+    osimModel->addJoint(&weld);
+    
+    // create an ExpressionBasedBushingForce that represents an
+    // uncoupled, linear bushing between the ball body and welded base body
+    Vec3 rotStiffness(0);
+    Vec3 transStiffness(stiffness);
+    Vec3 rotDamping(0);
+    Vec3 transDamping(0);
+    
+    ExpressionBasedBushingForce spring("base_body", Vec3(0), Vec3(0), "ball", Vec3(0), Vec3(0), transStiffness, rotStiffness, transDamping, rotDamping);
+    spring.setName("linear_bushing");
+    
+    osimModel->addForce(&spring);
+    
+    osimModel->print("ExpressionBasedBushingForceModel.osim");
+    
+    // Create the force reporter
+    ForceReporter* reporter = new ForceReporter(osimModel);
+    osimModel->addAnalysis(reporter);
+    
+    SimTK::State& osim_state = osimModel->initSystem();
+    
+    // set the initial height of the ball on slider
+    slider_coords[0].setValue(osim_state, start_h);
+    osimModel->getMultibodySystem().realize(osim_state, Stage::Position );
+    
+    //==========================================================================
+    // Compute the force and torque at the specified times.
+    RungeKuttaMersonIntegrator integrator(osimModel->getMultibodySystem() );
+    integrator.setAccuracy(1e-6);
+    Manager manager(*osimModel,  integrator);
+    manager.setInitialTime(0.0);
+    
+    double final_t = 2.0;
+    double nsteps = 10;
+    double dt = final_t/nsteps;
+    
+    for(int i = 1; i <=nsteps; i++){
+        manager.setFinalTime(dt*i);
+        manager.integrate(osim_state);
+        osimModel->getMultibodySystem().realize(osim_state, Stage::Acceleration);
+        Vec3 pos;
+        osimModel->updSimbodyEngine().getPosition(osim_state, ball, Vec3(0), pos);
+        
+        // compute the height based on the analytic solution for 1-D spring-mass
+        // system with zero-velocity at initial offset.
+        double height = (start_h-dh)*cos(omega*osim_state.getTime())+dh;
+
+        // check that the simulated solution is equivalent to the analytic solution
+        ASSERT_EQUAL(height, pos(1), 1e-4);
+        
+        // get the forces applied to the base and ball
+        Array<double> model_force = spring.getRecordValues(osim_state);
+        
+        // compute the expected force on the ball
+        double analytical_force = -stiffness*height;
+
+        // check analytical force corresponds to the force on the ball 
+        // in the Y direction, index = 7
+        ASSERT_EQUAL(analytical_force, model_force[7], 2e-4);
```

Comment:
Why 2e-4 here but 1e-4 above? Where did these tolerances come from and how do they relate to the integration accuracy?